modules/geolocation/geolocation.py

```diff
@@ -129,3 +129,35 @@ def calculate_pixel_to_geo_mapping(self):
 
         # Return matrix product of mappedGeoMatrix and sourcePixelMatrixInverse
         return (mappedGeoMatrix.dot(sourcePixelMatrixInverse))
+
+    def map_location_from_pixel(self, transformation_matrix, pixels):
+        """
+        Maps Geographical Location Coordinates in the destination image
+        
+        Parameters
+        -------
+            transformation_matrix : np.array(shape=(3,3))
+            pixels : np.array(shape=(5,2))
+
+        Returns
+        -------
+            np.array(shape=(5,2))
+        """
+
+        # Express all 2D coordinates of pixels as 3D coordinates with z value = 1
+        pixels = np.insert(pixels, 2, 1, axis = 1)
+
+        geoCoordinates = np.empty(shape=(0, 2))
+        
+        # Cycle through all 2D coordinates of pixels
+        for p in pixels:    
+            # Compute Homogeneous Coordinates: Product of Image Pixels and Coordinates
+            homogeneousCoordinates = transformation_matrix.dot(p)
+
+            # Dehomogenizing the coordinate vector to compute the position in the destination image
+            dehomogenizedX = homogeneousCoordinates[0] / homogeneousCoordinates[1]
+            dehomogenizedY = homogeneousCoordinates[1] / homogeneousCoordinates[2]
+   
```

Comment:
Where is this de-homogenizing equation defined? How is homogenousCoordinates a dot product, wouldn't transformation be achieved with a matrix-vector product which is not what np.dot does?

Reply:
I specifically followed steps 6 and 7 from this [post](https://math.stackexchange.com/questions/296794/finding-the-transform-matrix-from-4-projected-points-with-javascript).

As for the .dot() method, that can be used in numpy to perform matrix-vector multiplication. 

In `homogeneousCoordinates = transformation_matrix.dot(p)` I have multiplied the transformation matrix with the coordinates, to produce the homogeneous coordinates.

To be completely honest, I don't have a complete understanding of the math behind this, maybe Xierumeng can be of more help on that.

Reply:
According to documentation, `numpy.dot` is valid but it is preferred to use `numpy.matmul` or the `@` operator. See https://numpy.org/doc/stable/reference/generated/numpy.dot.html.

Reply:
Yeah please use numpy.matmul just to avoid confusion for future developers. As for the dehomogenization, I was just hoping to get some literature to verify its correctness, which you seem to have, so we're good.